lua/entities/npc_vj_creature_base/init.lua

```diff
@@ -2586,7 +2586,25 @@ function ENT:OnTakeDamage(dmginfo)
 	end
 	if self.Dead == true then DoBleed() return 0 end -- If dead then just bleed but take no damage
 	
-	self:CustomOnTakeDamage_BeforeDamage(dmginfo, hitgroup)
+	self:CustomOnTakeDamage_BeforeDamage(dmginfo, hitgroup) 
+	if dmginfo:GetDamage() >= 1 then self:SetCondition(17) end -- COND_LIGHT_DAMAGE 
+	if dmginfo:GetDamage() >= 20 then self:SetCondition(18) end -- COND_HEAVY_DAMAGE 
+	local dmgcount = self:GetSaveTable().m_iDamageCount 
+	local sumdamage = self:GetSaveTable().m_flSumDamage 
+	self:SetSaveValue("m_iDamageCount",dmgcount + 1) 
+	if self:GetSaveTable().m_flLastDamageTime > -1 then 
+		sumdamage = sumdamage + dmginfo:GetDamage() -- sum of all dmg in 1 sec 
+	else 
+		sumdamage = dmginfo:GetDamage() -- recount 
+	end 
+	self:SetSaveValue("m_flSumDamage",sumdamage) 
+	self:SetSaveValue("m_flLastDamageTime",0) 
+	if IsValid(dmginfo:GetAttacker()) then 
+		self:TriggerOutput("OnDamaged",dmginfo:GetAttacker()) 
+		if dmginfo:GetAttacker():IsPlayer() then self:SetSaveValue("m_flLastPlayerDamageTime",0) end 
```

Comment:
m_flLastPlayerDamageTime is used by the source function ent.GetLastPlayerDamageTime, which is only used by ai_playerally. Adding this to VJ Base has no use.